packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -254,15 +332,24 @@ export class AiAgentService {
             throw new ForbiddenError('Copilot is not enabled');
         }
 
-        // TODO:
-        // permissions
-
         const agent = await this.aiAgentModel.getAgent({
             organizationUuid,
             agentUuid,
             projectUuid,
         });
 
+        if (!agent) {
+            throw new NotFoundError(`Agent not found: ${agentUuid}`);
+        }
+
+        // Check group access
+        const hasAccess = await this.checkAgentGroupAccess(user, agent);
+        if (!hasAccess) {
+            throw new ForbiddenError(
+                'Insufficient permissions to access this agent',
+            );
+        }
```

Comment:
Should this check live under `getIsCopilotEnabled`?

Reply:
🤔 I need most of the info from the `agent` to be able to call this... do you have a different way to do it in the order you're suggesting?

Reply:
sorry, ignore this comment! I misread the method name